Add new --state option to allow key deletion by status

steveno opened the pull request:
### What does this PR do?
Adds a `--state` option to `salt-key` that allows you delete keys by what status they currently are.

e.g. `sudo salt-key --delete box1 --state accepted`

### What issues does this PR fix or reference?
https://github.com/saltstack/salt/issues/48951

### Previous Behavior
If there was a key with the same name in multiple statues it would delete all of them.

### Tests written?

No

### Commits signed with GPG?

Yes

steveno commented:
This is ready for review now. I can squash the commits if needed.

steveno commented:
Gentle ping.

steveno commented:
@rallytime This PR is ready for review

Commit:
compare probe/src directly issues/49855

This changes the tokenized approach for string comparison
and shifts it to direct comparison of the content and the line in question
to determine if any changes should be made during file.line insert/ensure before or after

jubrad committed:
file.py fix newlines for pylint

jubrad committed:
file.py one more pylint white space fix

jubrad committed:
strip newlines when checking string equality

Commit:
replace module.file._starts_till with _line_match

        new method matches lines with 0,1 bash like
        return codes

        method no longer indicates where in a line a
        match occurs but compares the source and probe
        directly

steveno commented:
@rallytime or @twangboy Is there anything else I need to do for this to be merged?

Ch3LL reviewed (changes requested):
thanks for adding this feature :) would you mind adding some tests so we make sure this new feature doesn't regress in the future. Thanks!

steveno commented:
@Ch3LL Sorry about the delay in getting back to you. I'll to drum up some tests this week before pycon.

steveno commented:
Adding tests for this has been more difficult for me than expected. In both terms of "how" and "the test runner hates me". I apologize for the delay. For the record, I am still working on this.

Ch3LL commented:
@steveno no worries. anything i can help with?

Ch3LL commented:
for the rejected key: `key = os.path.join(pki_dir, 'minions', min_name)` you will want to make sure thats in the `minions_rejected` directory. So should look like this `key = os.path.join(pki_dir, 'minions_rejected', min_name)`

```
        # Make sure the accepted key is still present
        check_key = self.run_key('-p {0}'.format(min_name))
        self.assertEqual([], check_key)
```

I think you actually want the accepted key to be in the accepted keys so the return value will not be empty here, it will contain the `minion_dup` accepted key and should not contain the rejected key. 

Let me know if that helps or not :)

s0undt3ch committed:
Switch docs building under Py3

Refs #53137

s0undt3ch committed:
Have sphinx turn errors into warnings

s0undt3ch committed:
Allow docs to be built on Python >= 3.5, not just Python 3.6

steveno commented:
@Ch3LL I modified the test with your suggestions. I was able to get it working. Thank you very much for your assistance!!

This is ready for review once again.

Ch3LL commented on salt/key.py:
```diff
@@ -340,6 +344,17 @@ def _get_key_attrs(self, keydir, keyname,
         return keydir, keyname, keysize, user
 
+    def _get_selected_state(self, selected_state=''):
+        if selected_state.startswith('acc'):
```
I'm not 100% familiar with this code, but I'm wondering if we want to validate the users input to ensure they typed out the entire word and spit out an error telling them they did not include a valid state if they mispell it, instead of just checking it with `startswith`. what do you think @DmitryKuzmenko since you have been in this code before ? Any chance you can review this PR and what do you think about my suggestion?

steveno replied:
That if statement was "copied" from line 568. If you want to change it in my PR we should probably change it in both places.

```python
def list_status(self, match):
        '''
        Return a dict of managed keys under a named status
        '''
        acc, pre, rej, den = self._check_minions_directories()
        ret = {}
if match.startswith('acc'):
...
```

Ch3LL replied:
ahh thanks for pointing out! might not be a concern then. will wait to hear from @DmitryKuzmenko

Ch3LL commented:
thanks for following up with that test I really appreciate it. i just left one possible suggestion and also want to get @DmitryKuzmenko feedback before merge. Thanks!

Ch3LL committed:
Add tests for #52333

aplanas committed:
systemd_service: add firstboot function

systemd-firstboot is a command line tool from systend that will
set some basic configuration options, like the locale, keyboard
configuration, or the machine-id of the system.

Ch3LL commented:
FYI I migrated this PR from develop to neon to ensure it is included in the upcoming neon release. Let me know if this caused any issues. Thanks

jubrad committed:
strip probe as well as src in _line_match

dwoz committed:
Merge pull request #53409 from s0undt3ch/hotfix/fix-pkg-reqs-develop

[develop] Fix static requirements

twangboy committed:
Merge branch 'develop' into fix_win_network

Akm0d committed:
Merge branch 'develop' into etcd-returner-is-busted

amendlik committed:
Add support for Javadoc-style banners in slsutil.banner

twangboy committed:
Merge pull request #53399 from twangboy/fix_win_network

Fix test suite on develop

s0undt3ch committed:
Merge branch 'develop' into hotfix/fix-nox-bypass-develop

amendlik committed:
Add example documentation for slsutil.banner

amendlik committed:
Handle exception in slsutil.banner where width is too small to render

s0undt3ch committed:
Merge pull request #53487 from s0undt3ch/hotfix/fix-nox-bypass-develop

[develop] Workaround nox's install only flag

twangboy committed:
Merge branch 'develop' into add-fully-qualified-cmd-call

twangboy committed:
Merge pull request #53311 from doesitblend/add-fully-qualified-cmd-call

Add fully qualified cmd call

Akm0d committed:
Merge branch 'develop' into etcd-returner-is-busted

Akm0d committed:
Merge pull request #51363 from arizvisa/etcd-returner-is-busted

The etcd returner is using the wrong API due to a refactor of salt.utils.etcd_util.

Akm0d committed:
Merge branch 'develop' into patch-1

mattp- committed:
add general exception catch in salt.utils.store_job

external returners can blow up for any variety of reasons, but when we
don't catch it it will sometimes stop the event from being returned to
the bus via fire_event, which has downstream effects. so lets guard
against that.

Commit:
Add support for Heimdal Kerberos

Commit:
Fix list principals and get privs

mattp- committed:
lint

Akm0d committed:
Removed trailing whitespace

Fixes pylint errors

garethgreenaway committed:
Merge pull request #52856 from steveno/patch-1

Add note about new flatpak module

Akm0d committed:
Merge branch 'develop' into support-heimdal-krb

garethgreenaway committed:
Merge pull request #53517 from bloomberg/returner-tighten

add some exception guarding in salt.utils.job

Akm0d committed:
Merge branch 'develop' into support-heimdal-krb

Commit:
Merge branch 'develop' into develop-fedora29

Akm0d committed:
Merge branch 'develop' into fix_ipaddress_python3

Akm0d committed:
Merge pull request #51803 from misch42/fix_ipaddress_python3

Fix for issue #51789 import syntax in python3

Akm0d committed:
Merge branch 'develop' into support-heimdal-krb

Akm0d committed:
Merge pull request #53515 from mickenordin/support-heimdal-krb

Support Heimdal Kerberos

Akm0d committed:
Added test for issue #4438

Akm0d committed:
Merge branch 'develop' of https://github.com/saltstack/salt into pkgrepo

Akm0d committed:
Check if key_url needs changing

Akm0d committed:
Removed OS specification

Akm0d committed:
Merge branch 'develop' into patch-2

Akm0d committed:
Removed unused import and unsolicited formatting

Akm0d committed:
Fixed formatting

benjumanji committed:
Don't double encode s3 paths

As s3.query already knows how to quote paths, don't quote the passed in
path: s3.query will handle it.

Akm0d committed:
Merge branch 'develop' into patch-1

Akm0d committed:
Created separate tests

Akm0d committed:
enable testing mode for these cases

Akm0d committed:
Update salt/states/pkgrepo.py

Co-Authored-By: Wayne Werner <waynejwerner@gmail.com>

Commit:
Merge branch 'develop' into develop-fedora29

waynew committed:
Merge pull request #53537 from Akm0d/pkgrepo

Check if the key_url changed in pkgrepo.managed

Akm0d committed:
Merge branch 'develop' into patch-2

Akm0d committed:
Merge pull request #52798 from myii/patch-2

docs(config.py): fix order of `config.get`

Akm0d committed:
Merge branch 'develop' into patch-1

Akm0d committed:
Merge branch 'develop' into feature/tplroot

Commit:
Merge branch 'develop' into develop-fedora29

Akm0d committed:
Merge pull request #51814 from myii/feature/tplroot

feat(tpldata): provide `tplroot`

Akm0d committed:
Merge branch 'develop' into develop

Akm0d committed:
Merge pull request #51572 from kaidokert/develop

Fix error retry handling for Route53

Akm0d committed:
Merge branch 'develop' into infra5209

Akm0d committed:
Merge pull request #51923 from bodhi-space/infra5209

INFRA-5209 - Avoid throwing exception for missing security group in test mode

Akm0d committed:
Merge branch 'develop' into issue-51957

Akm0d committed:
Merge pull request #51965 from recursiveMake/issue-51957

Remove package name prefix from member.name

Akm0d committed:
Merge branch 'develop' into patch-1

Ch3LL committed:
Merge pull request #51670 from OrangeDog/patch-4

 Fix extract_hash case for top-level binary shasums

mattp- committed:
catch any exception on reading rsa key

an unhandled exception thrown at this point of code will put MWorker
into an unrecoverable hung state that leaks memory as MWorkerQueue
continues passing it work that is ignored. In my case I encountered an
RSAError but I see no reason to be specific for exception class here.

Akm0d committed:
Merge branch 'develop' into patch-1

mattp- committed:
add debug exc

mattp- committed:
log trace on debug loglevel

Commit:
Merge branch 'develop' into develop-fedora29

Commit:
Merge pull request #53427 from bryceml/develop-fedora29

add fedora 29 tests to pr's

s0undt3ch committed:
Update pipelines to work on old and new jenkins

s0undt3ch committed:
Merge pull request #53528 from s0undt3ch/hotfix/fix-nox-bypass-develop

[develop] Update pipelines to work on old and new jenkins

s0undt3ch committed:
Merge pull request #53555 from bloomberg/auth-rsa

catch any exception on reading rsa key

yohonet committed:
Update pillar.py

I believe something's missing in the documentation.

s0undt3ch committed:
Minor fixes/adjustments to the new CI pipelines

cmcmarrow committed:
Merge pull request #53571 from yohonet/patch-1

Update pillar.py

cro committed:
Forward port PR 53412 to Neon

cro committed:
Make doc build test pass

cro committed:
Merge pull request #53589 from cro/53412-develop

Forward port PR 53412 to Develop

s0undt3ch committed:
Merge pull request #53578 from s0undt3ch/hotfix/fix-nox-bypass-develop

[develop] Minor fixes/adjustments to the new CI pipelines

Commit:
Adds tests for states/win_iis.py

s0undt3ch committed:
CI Pipelines. Timeout after getting a node. Report exit code.

Akm0d committed:
Merge branch 'develop' into patch-1

Commit:
Adds tests for win_iis

cmcmarrow committed:
Merge branch 'develop' into state_test_win_iis

cmcmarrow committed:
Merge pull request #53605 from cmcmarrow/state_test_win_iis

Adds tests for states/win_iis.py

Akm0d committed:
Merge branch 'develop' into patch-1

cro committed:
__opts__ is not self.opts for metaproxies.

cro committed:
Add unit test to check for __opts__ vs opts mismatch

DmitryKuzmenko committed:
Merge pull request #53617 from cro/metaproxy_matcher_opts_develop

Fix typo, __opts__ should be opts

sathieu committed:
runners.git_pillar: Also match the repo parameter against the repo name

Akm0d committed:
Create a state for the cpan module (#53413)

* Improved stability of cpan module
- Added command line options for cpan module installation
- Give many more details when things do change
- Parse cpan config using literal_eval, which is cleaner and more stable
  than the line-by-line parsing
- rename 'show_config' to 'config' to remove redundancy
- Rebuilt all command strings as lists to facilitate adding more options in the future
- added a "get cpan binary" function to facilitate expansion to cpanm
  and cpanp in the future
* Solves https://github.com/saltstack/salt/issues/2130
- cpan module returns version number
- created cpan state module
* Allow mirrors to be passed as a comma-separated string or as a list
* Install a cpan module that has no dependencies
* Added cpan state to docs, correct name for centos pkg

Akm0d committed:
Merge branch 'develop' into patch-1

s0undt3ch committed:
Merge pull request #53612 from s0undt3ch/hotfix/fix-nox-bypass-develop

CI Pipelines. Timeout after getting a node. Report exit code.

Akm0d committed:
Merge branch 'develop' into patch-1

dhiltonp committed:
generify masterminion returners function not found error - may be because returner isn't loaded

Akm0d committed:
Merge branch 'develop' into git_pillar_update_by_name

xeacott committed:
Merge branch 'develop' into photon_pkgrepo

Akm0d committed:
Merge pull request #53628 from dhiltonp/gen

Provide more information on the returner's "function not found error"

Akm0d committed:
Merge branch 'develop' into patch-1

dhiltonp committed:
kernel, os, os_family may not exist in __grains__ - use get

dhiltonp committed:
look at os_family['Solaris'] instead of checking all* solaris distros

waynew committed:
Refactor & test line, fix win issues

dhiltonp committed:
Merge branch 'develop' into some-pkgin

Akm0d committed:
Merge branch 'develop' into git_pillar_update_by_name

Akm0d committed:
Merge pull request #53622 from sathieu/git_pillar_update_by_name

runners.git_pillar: Also match the repo parameter against the repo name

Akm0d committed:
Merge branch 'develop' into patch-1

garethgreenaway committed:
Merge branch 'develop' into some-pkgin

dhiltonp committed:
Merge pull request #53632 from dhiltonp/some-pkgin

pkgin is on Solaris...

garethgreenaway committed:
Merge pull request #52550 from lomeroe/photon_pkgrepo

Add support for VMware Photon OS to pkgrepo

waynew committed:
Improve file.line docs

garethgreenaway committed:
Create test_s3fs.py

Adding test_get_file_from_s3 to ensure that path argument is not double quoted.

garethgreenaway committed:
Merge branch 'develop' into antiquote

waynew committed:
Fix some tests

Ch3LL committed:
Merge branch 'develop' into banner

garethgreenaway committed:
Update test_s3fs.py

Adding test to ensure paths with spaces are properly quoted.

waynew committed:
Fix lint

s0undt3ch committed:
Add Fedora 30 requirements files

s0undt3ch committed:
Python 2 is no longer supported

s0undt3ch committed:
Delete Fedora 28 static requirements

s0undt3ch committed:
Delete Opensuse 42 static requirements

Commit:
adds windows event log support

s0undt3ch committed:
Switch docs and lint to Py3

s0undt3ch committed:
Fix doc build errors

Commit:
adds win_event_viewer

Commit:
adds win_event_viewer

Commit:
adds win_evnet_viewer

Ch3LL committed:
Update CODEOWNERS github file

waynew committed:
Fallback to no UID

If we can't figure it out, we shouldn't error out.

Commit:
adds win_event_viewer

Commit:
adds win_event_viewer

Commit:
adds win_event_viewer

garethgreenaway committed:
Merge pull request #53540 from benjumanji/antiquote

Don't double encode s3 paths

dbuenor committed:
Fixes bug in MongoDb User State causing changes to be applied in case of an existing user even in test mode.

s0undt3ch committed:
Stop using the deprecated `log.warn()` method

s0undt3ch committed:
Upgrade to latest PyLint

s0undt3ch committed:
Stop using deprecated methods(actually remove py2 only code)

s0undt3ch committed:
Enhance lint report

s0undt3ch committed:
Use milestones to abort previous, still running, builds, on new builds

s0undt3ch committed:
Merge pull request #53699 from s0undt3ch/features/f30-reqs-develop

[develop] Add Fedora 30 requirements files

waynew committed:
Skip chattr tests on Windows

Undoes the previous commit that was fixing a failure on Windows.

s0undt3ch committed:
Archive the kitchen logs

s0undt3ch committed:
Improve slack message

Akm0d committed:
Merge branch 'develop' into patch-1

Commit:
adds win_event_viewer

Commit:
adds win_event_viewer

Commit:
adds win_event_viewer

Commit:
adds win_event_viewer

s0undt3ch committed:
Merge pull request #53727 from s0undt3ch/features/f30-reqs-develop

[develop] Archive the kitchen logs

Commit:
adds win_event_viewer

Commit:
adds win_event_viewer

Commit:
adds win_event_viewer

Commit:
adds win_event_viewer

Commit:
adds win_event_viewer

Commit:
adds win_event_viewer

Commit:
adds win_event_viewer

Akm0d committed:
Don't call potentially missing shade library (#53734)

* Check for openstack

* Account for HAS_SHADE not being a tuple

Commit:
adds win_events_viewer

Commit:
adds win_event_viwer

Commit:
adds win_event_viewer

Commit:
adds win_event_viewer

Akm0d committed:
Merge branch 'develop' into patch-1

Commit:
adds win_event_viewer

Commit:
adds win_event_viewer

cmcmarrow committed:
Merge branch 'develop' into win_event_viewer

cmcmarrow committed:
Merge pull request #53700 from cmcmarrow/win_event_viewer

adds windows event log support

sjorge committed:
py3 results in broken mdata grains

When running salt on SmartOS with py3, the mdata grains are broken.
Both stem from the same underlying issue that stderr is mixed in
 with stdout for the `mdata-list` and `mdata-get`. (Is this new?)

The solution is to ignore the output if it starts with 'ERROR:'.
I couldn't find a way to filter out stderr via cmd.run directly.

sjorge committed:
Missing grains on SmartoS GZ under py3

Some grains were broken in the GZ too,
 also do some mindor cleanup.

1. /etc/pkgsrc_version
Does not exist in GZ and is a non-supported interface for zones,
 we can get this information from the pkg_install.conf file too.
This should also more accuratly reflect the correct value.

The path for the file is different in the GZ so we need to support both locations

2. pkgin repos are missing in GZ only
Same as above, the path is different so we need to support both.

sjorge committed:
py3 salt.utils.platform.is_smartos_zone/global fix

The output of Popen is bytes not string so the checks below for 'global' was not matching.

- cleanup the Popen bits a bit
- decode('utf8') the output

Commit:
fix names to various package modules so links work

Commit:
increase sleep time between kitchen create failures to account for api limits

dwoz committed:
Merge pull request #53876 from bryceml/develop-increase-sleep

increase sleep time between kitchen create failures to account for api limits

s0undt3ch committed:
Update CI pipelines. Remove old jenkins cruft.

waynew committed:
Merge pull request #53889 from waynew/49855-fix-file.line-issues-develop

49855 fix file.line issues develop

dbuenor committed:
Added new test for mongodb_user.present that verifies the proper functioning when aplying state for an existing user

sjorge committed:
Add tests for salt.grains.mdata

sjorge committed:
Add tests for salt.grains.smartos

dbuenor committed:
Fixed whitespaces at mongodb_user.present test #53963

nikolas committed:
Fix typo in vault docs

atleast -> at least

Ch3LL committed:
Merge pull request #53893 from Ch3LL/codeowners_develop

[develop] Update CODEOWNERS github file

Commit:
Fix a typo in zonecfg module

The zone net resource properties vlan-id and defrouter were combined into a single array element, which prevented Salt from setting either of them.  This commit splits them into their correct seperate elements and allows Salt to correctly set the properties.

Akm0d committed:
Merge pull request #53975 from ninefathom/zonecfg-fix-001

Fix a breaking typo in zonecfg module

eliasp committed:
auth: log details in case of user auth issue

When a local SaltClient sends a job as "root" while the Master runs as a non-root user, make it more clear why the message `Authentication failure of type "user" occurred.` is logged.

eliasp committed:
tests: fix typo (`perkissive` > `permissive`)

dbuenor committed:
Changed returned result when using test=true for mongodb_user.present to avoid false positives.

dbuenor committed:
Refactorized mongodb_user.present to avoid showing successful changes when no changes has been made to user.
Now shows changes when roles has been changed for user so its easier to see if roles going to changed in test mode.
Changed tests to adapt to new behaviour.

waynew committed:
Merge pull request #53994 from eliasp/develop-typo-perkissive-permissive

tests: fix typo (`perkissive` > `permissive`)

eliasp committed:
auth: use log formatting for message

Co-Authored-By: Wayne Werner <waynejwerner@gmail.com>

waynew committed:
Fix potential recursion/duplicate object detection

It's unlikely, but possible, that we could get a problem with over-eager
detection of recursion. Especially in the cases of short interned
strings or small numbers on CPython, since those are cached. Instead
we'll just check of it's a type that we're going to recurse on.

aplanas committed:
zypperpkg: fix pkg.list_pkgs cache

The cache from pkg.list_pkgs for the zypper installer is too aggresive.
Some parameters will deliver different package lists, like root and
includes. The current cache do not take those parameters into
consideration, so the next time that this function is called, the last
list of packages will be returned, without checking if the current
parameters match the old one.

This patch create a different cache key for each parameter combination,
so the cached data will be separated too.

waynew committed:
Merge pull request #54024 from aplanas/fix_zypperpkg

zypperpkg: fix pkg.list_pkgs cache

waynew committed:
Merge pull request #54018 from waynew/more-recursive-msgpack-fix

Fix potential recursion/duplicate object detection

waynew committed:
Merge pull request #53381 from aplanas/fix_firstboot

systemd_service: add firstboot function

Commit:
improve documentation regarding state.sls

eliasp committed:
auth: correct typos in docstrings

waynew committed:
Merge pull request #54047 from eliasp/salt.auth-typos

auth: correct typos in docstrings

Akm0d committed:
Merge branch 'develop' into patch-1

Commit:
Merge branch 'develop' into state-sls-docs

dbuenor committed:
Commit to fix lint problems to allow successful pr-lint build

Ch3LL committed:
Merge branch 'develop' into banner

Natrinicle committed:
Adding accept_ra 2 option to modules.debian_ip

Currently, the Debian IP module only allows 0 or 1 but 2 is a valid
and sometimes required option.

```
accept_ra - INTEGER
	Accept Router Advertisements; autoconfigure using them.

	It also determines whether or not to transmit Router
	Solicitations. If and only if the functional setting is to
	accept Router Advertisements, Router Solicitations will be
	transmitted.

	Possible values are:
		0 Do not accept Router Advertisements.
		1 Accept Router Advertisements if forwarding is disabled.
		2 Overrule forwarding behaviour. Accept Router Advertisements
		  even if forwarding is enabled.

	Functional default: enabled if local forwarding is disabled.
			    disabled if local forwarding is enabled.
```
https://www.kernel.org/doc/Documentation/networking/ip-sysctl.txt

Commit:
adds webconfig tests

amendlik committed:
Update slsutil.banner example documentation

waynew committed:
Merge pull request #53794 from sjorge/py3_smartos_grains

Fixes SmartOS grains when using py3

Ch3LL committed:
Merge branch 'develop' into patch-1

Ch3LL committed:
Merge pull request #51628 from myii/patch-1

Update `formulas.rst` with new IRC channel and links to IRC logs

Natrinicle committed:
Adding Unit Tests and Fixing debian_eth.jinja

Had to undo the replace all underscores with dashes in the debian_eth
template as that messes up accept_ra which needs to be an underscore
to work.

Added a Unittest as "eth20" for testing accept_ra and autoconf 1 in
IPv6 interfaces.

cmcmarrow committed:
Merge branch 'develop' into web_confi_test

Commit:
Merge branch 'develop' into develop

Commit:
updating git repo where rbenv is hosted

Commit:
Merge branch 'develop' into state-sls-docs

dwoz committed:
Merge pull request #54144 from bryceml/update_rbenv_git_repo

updating git repo where rbenv is hosted

Commit:
Merge branch 'develop' into develop

dwoz committed:
Merge pull request #54037 from bryceml/state-sls-docs

improve documentation regarding state.sls

Commit:
Merge branch 'develop' into develop

Commit:
Fix x509 typos

Ch3LL committed:
Fix grains.core pylint error

dwoz committed:
Merge pull request #54158 from stratusjerry/typo_fix

Fix x509 typos

dwoz committed:
Merge pull request #53499 from amendlik/banner

Updates to slsutil.banner

Commit:
Merge branch 'develop' into develop

Natrinicle committed:
Merge branch 'develop' into add-accept-ra-option

Commit:
add new saltconf ads

dwoz committed:
Merge pull request #54255 from bryceml/develop_new_saltconf_ads

add new saltconf ads

Commit:
Merge branch 'develop' into develop

dwoz committed:
Merge pull request #53860 from bryceml/develop

fix names to various package modules so links work

steveno commented:
Is there anyway this can get reviewed/merged? It's been open since February at this point.

Ch3LL commented:
bump @DmitryKuzmenko can you review here please?

Ch3LL commented:
@steveno just fyi we have been slowing down merges to try to stabilize the test suite, especially on neon. Theres some more details here https://groups.google.com/forum/#!topic/salt-announce/5RJjtDYszx8

but i'll try to get this ready so we can just merge when ready :)

Natrinicle committed:
Merge branch 'develop' into add-accept-ra-option

DmitryKuzmenko commented on salt/key.py:
```diff
@@ -474,7 +489,10 @@ def name_match(self, match, full=False):
             match = match.split(',')
+        selected_state = self._get_selected_state(self.opts['state'])
         for status, keys in six.iteritems(matches):
+            if selected_state != '' and status != selected_state:
```
Could you please replace `if selected_state != ''` with `if selected_state`?

steveno replied:
Fixed.

DmitryKuzmenko commented on salt/key.py:
```diff
@@ -693,8 +711,12 @@ def delete_key(self,
+        selected_state = self._get_selected_state(self.opts['state'])
         for status, keys in six.iteritems(matches):
             for key in keys:
+                if selected_state != '' and status != selected_state:
```
The same is here.

steveno replied:
Fixed.

DmitryKuzmenko reviewed (changes requested):
This looks okay except one minor change request.

cmcmarrow committed:
Merge branch 'develop' into web_confi_test

dwoz committed:
Merge pull request #53964 from nikolas/vault-docs-typo

Fix typo in vault docs

dwoz committed:
Merge pull request #54163 from Ch3LL/lint_core

Fix grains.core pylint error

dwoz committed:
Merge pull request #53965 from dbuenor/dbueno/bug-mongodb-user-present-aplying-changes-on-existing-user-even-if-testing-rebased

Mongodb_user.present aplying changes on existing mongodb user even if Test=true

dwoz committed:
Merge pull request #54069 from cmcmarrow/web_confi_test

adds webconfig tests

dwoz committed:
Merge pull request #53990 from eliasp/develop-log-warning-user-auth-error

auth: log details in case of user auth issue

dwoz committed:
Merge pull request #54067 from Natrinicle/add-accept-ra-option

Adding accept_ra 2 option to modules.debian_ip

sblaisot committed:
Import abstract base classes from collection.abc in python 3.3+

Commit:
modifying saltconf ads

dwoz committed:
Merge pull request #54363 from bryceml/develop_saltconf_ads

modifying saltconf ads

Commit:
wrap codecov in || true

dwoz committed:
Merge pull request #53921 from s0undt3ch/hotfix/ci-pipelines-develop

[develop] Update CI pipelines. Remove old jenkins cruft.

Commit:
change to ruby 2.6.3

Commit:
Merge pull request #54708 from bryceml/develop_ruby

switch to ruby 2.6.3

Commit:
Update release versions for develop branch

Commit:
Merge pull request #54745 from frogunder/doc_conf_develop

Update release versions for develop branch

DmitryKuzmenko committed:
Merge pull request #54309 from sblaisot/collections-deprecated

Import abstract base classes from collection.abc in python 3.3+

DmitryKuzmenko committed:
Revert "Import abstract base classes from collection.abc in python 3.3+"

DmitryKuzmenko committed:
Merge pull request #54832 from saltstack/revert-54309-collections-deprecated

Revert "Import abstract base classes from collection.abc in python 3.3+"

Commit:
Update release versions for the develop branch

dwoz committed:
Merge pull request #55093 from frogunder/doc_conf_develop

Update release versions for the develop branch

steveno commented:
@Ch3LL or @DmitryKuzmenko Now that 2019.2.2 has been released, can we take another look at this?

DmitryKuzmenko commented:
@steveno Sorry for delay. Since we have new branching strategy could you please rebase your PR to `master` branch and ping me again?

dwoz commented:
@steveno Looks like this needs to be rebased against master again.

dwoz commented:
@steveno this branch looks a bit messed up. Can you please point me to the original commit ref?

steveno commented:
@dwoz as you pointed out this MR is a mess now. I don't even know what the original commits where or how to find them.

steveno commented:
@dwoz Is there interest in actually merging this now? A year later, I forgot I even did this.

waynew commented:
https://github.com/saltstack/salt/commit/f6d1a1b6cc6f3a3dc36dfcc73064b4bbed5434f6 should be the original ref, or at least contain the older commits

steveno commented:
@waynew Nice!

My original commits were (in chronological order):
6c4293885520e5caf80633d2e33124615b6c332b
e6c3affcbfceff5091865687ca36955871c1802f
55fa695fea110f4fce49e16f9cdc886820832318
c8b8bc7a1e9c16251688706b917bd24bd758356f
035c8c2c7b3eb6764787a1ee96e4f3feeb8adbd3
d295a338752b54ae2dd844784742fd42022f18ba

steveno commented:
I'm just going to open a new MR.

mdschmitt commented:
@steveno , is there a link to that new MR, or is this one of those "lost in the shuffle & forgot about" kind of MRs?

steveno commented:
> @steveno , is there a link to that new MR

No.

> is this one of those "lost in the shuffle & forgot about" kind of MRs?

No. This MR was closed right around when they stopped accepting MRs so they could rework the test suite and such. Then when they started accepting MRs again the number of open MRs magically seemed to skyrocket and, well, I lost interest by then.
